### PR TITLE
mlflow 2.0 compatibility

### PR DESCRIPTION
**Pull request by shrinath-suresh**
Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

## What changes are proposed in this pull request?

MLflow 2.0 compatibility with latest PT and PTL

## How is this patch tested?

Titanic
======
Screenshot from 2022-12-15 08-25-58
Screenshot from 2022-12-15 08-26-42
Screenshot from 2022-12-15 08-27-19
[titanic_training.txt](https://github.com/mlflow/mlflow-torchserve/files/10233992/titanic_training.txt)

Iris Classification
=============
[iris_training.txt](https://github.com/mlflow/mlflow-torchserve/files/10233993/iris_training.txt)
[iris_training_gpu.txt](https://github.com/mlflow/mlflow-torchserve/files/10233995/iris_training_gpu.txt)
Screenshot from 2022-12-15 09-51-51
Screenshot from 2022-12-15 09-52-08
Screenshot from 2022-12-15 09-52-27

Screenshot from 2022-12-15 09-52-44

IrisClassification - Torchscripted
=========================
[iris_torchscript_training.txt](https://github.com/mlflow/mlflow-torchserve/files/10233996/iris_torchscript_training.txt)
Screenshot from 2022-12-15 10-01-34
Screenshot from 2022-12-15 10-01-49
Screenshot from 2022-12-15 10-02-17

MNIST
======
[mnist_training.txt](https://github.com/mlflow/mlflow-torchserve/files/10233999/mnist_training.txt)
[mnist_training_gpu.txt](https://github.com/mlflow/mlflow-torchserve/files/10234000/mnist_training_gpu.txt)
Screenshot from 2022-12-15 10-16-03
Screenshot from 2022-12-15 10-16-22

CIFAR
=====
[cifar_training.txt](https://github.com/mlflow/mlflow-torchserve/files/10234001/cifar_training.txt)
[cifar_training_gpu.txt](https://github.com/mlflow/mlflow-torchserve/files/10234002/cifar_training_gpu.txt)
Screenshot from 2022-12-15 10-32-13
Screenshot from 2022-12-15 10-33-48

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow TorchServe Deployment Plugin users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?
Components
- [ ] `area/deploy`: Main deployment plugin logic
- [ ] `area/build`: Build and test infrastructure for MLflow TorchServe Deployment Plugin
- [ ] `area/docs`: MLflow TorchServe Deployment Plugin documentation pages
- [x] `area/examples`: Example code

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/IrisClassification/iris_classification.py [hunk 1]
```diff
@@ -21,9 +21,9 @@ class IrisClassification(pl.LightningModule):
     def __init__(self, **kwargs):
         super(IrisClassification, self).__init__()
 
-        self.train_acc = Accuracy()
-        self.val_acc = Accuracy()
-        self.test_acc = Accuracy()
+        self.train_acc = Accuracy(task="multiclass", num_classes=3)
```
Comment (shrinath-suresh) [inline examples/IrisClassification/iris_classification.py:24]: Latest torchmetrics expects `task` and `num_classes` to be passed while using `Accuracy` - https://torchmetrics.readthedocs.io/en/stable/classification/accuracy.html

File: mlflow_torchserve/__init__.py [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
+
+from mlflow_torchserve.config import Config
 
 _logger = logging.getLogger(__name__)
 
 _DEFAULT_TORCHSERVE_LOCAL_INFERENCE_PORT = "8080"
 _DEFAULT_TORCHSERVE_LOCAL_MANAGEMENT_PORT = "8081"
 
 
+class CustomPredictionsResponse(PredictionsResponse):
```
Comment (dbczumar) [inline mlflow_torchserve/__init__.py:25]: Nit: Do we want to call this `TorchServePredictionsResponse`?

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @shrinath-suresh !

**Comment by Chichilele** [comment]
Hi, are you still considering this PR?

**Comment by shrinath-suresh** [comment]
@chauhang Can you please review and approve the PR so that we can merge ?

**Review by chauhang — APPROVED** [review]

**Comment by shrinath-suresh** [comment]
Thanks @chauhang and @dbczumar

File: mlflow_torchserve/__init__.py (hunk 2 above)
Comment (shrinath-suresh) [reply mlflow_torchserve/__init__.py:25]: @dbczumar I have done this change as a separate [PR](https://github.com/mlflow/mlflow-torchserve/pull/98). Can you please review it when you find time. The builds are passing.

**Pull request merged**